src/binary_annotator.cpp

```diff
@@ -0,0 +1,768 @@
+#include "binary_annotator.h"
+
+#include <iostream>
+#include <vector>
+
+#include "flatbuffers/reflection.h"
+#include "flatbuffers/verifier.h"
+
+namespace flatbuffers {
+namespace {
+
+static BinaryRegion MakeBinaryRegion(
```

Comment:
optional style nit: You may as well use array initialization at call sites instead of this function. I don't think the factory function adds anything. Same with MakeBinarySection

Reply:
I was having a hard time appeasing the Windows checks with this. I had array initialization at first, but they don't work with emplace_back on windows... So I just made them these factory methods which do work with emplace_back.